### PR TITLE
Fix OIDC and e2e tests

### PR DESCRIPTION
**Pull request by squat**
This PR fixes the OIDC configuration so that the thanos-rule-syncer
can properly authenticate with the Observatorium API. Previsouly, the
OIDC configuration was never applied to the HTTP client's underlying
transport, meaning that it had no effect and all requests would fail.

This PR also makes the interval at which the Observatorium API is
polled configurable.

Finally, this PR fixes the thanos-rule-syncer so that it no longer
uses hard-coded paths for the Observatorium CA.

This PR also completes the e2e test scenario. In the end, the tests
run a semi-realistic scenario with an Observatorium API
exposing rules using a file as the rule repository, a thanos-rule-syncer
configuring a Thanos Ruler, and a Thanos Querier evaluating the rules
all being authenticated via Hydra.

### CODE REVIEW

File: go.mod
Comment (squat) [inline go.mod:16]: temporary until https://github.com/observatorium/api/pull/138 merges

**Comment by squat** [comment]
cc @allenmqcymp @metalmatze @observatorium/maintainers

**Comment by bwplotka** [comment]
image

 😢

**Comment by squat** [comment]
Yes, that's due to the tooling dependencies. We can move the tooling to Bingo in a followup to remove those things. For the purposes of review, only the first two PRs are important; the last is a tooling change to vendor the Observatorium  API.

**Review by bill3tt — APPROVED** [review]
LGTM assuming integration tests pass on your local machine :+1:

**Review by bwplotka — APPROVED** [review]

**Pull request merged**
